### ISSUE TITLE
Make `r.union` preserve selections

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4986]
This is a follow-up of https://github.com/rethinkdb/rethinkdb/issues/4981 . Something like this should probably work:
```js
r.union(
  r.table('devices').getAll(id, { index: 'id' }),
  r.table('relations').getAll(id, { index: 'subject' }),
  r.table('relations').getAll(id, { index: 'observer' })
).delete()
```

**Comment by marshall007** [comment]
> Would this also apply when selecting tables themselves?
> 
> ```js
> r.union(
>   r.table('tasks'),
>   r.table('sources')
> ).indexCreate('customer_id')
> 
> r.union(
>   r.table('tasks'),
>   r.table('sources')
> ).getAll(id, { index: 'customer_id' })
> ```

> The `index` example I think would be a separate step. Do you think this would actually be useful compared to putting the `getAll` inside the `union`?

I don't think it would be that useful except maybe in a database-per-customer scenario where all the tables are the same. I was mostly asking to see if things like that would be an emergent property of maintaining selections in `r.union`. So I don't think it's worth explicitly supporting if that's not the case.
